sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/model/ValueMappingModel.java

```diff
@@ -0,0 +1,44 @@
+package org.sdase.commons.keymgmt.model;
+
+import java.util.*;
+
+@SuppressWarnings("unused")
+public class ValueMappingModel {
+
+  private final Map<String, String> apiToImplBidirectional = new HashMap<>();
+  private final Map<String, String> implToApiBidirectional = new HashMap<>();
+
+  private final Map<String, String> implToApi = new HashMap<>();
+  private final Map<String, String> apiToImpl = new HashMap<>();
+
+  public ValueMappingModel setApiToImplBidirectional(List<ValueMapping> apiToImplBidirectional) {
+    apiToImplBidirectional.forEach(
+        vm -> {
+          this.apiToImplBidirectional.put(vm.getApi(), vm.getImpl());
+          this.implToApiBidirectional.put(vm.getImpl(), vm.getApi());
+        });
+    return this;
+  }
+
+  public ValueMappingModel setImplToApi(List<ValueMapping> implToApi) {
+    implToApi.forEach(vm -> this.implToApi.put(vm.getImpl(), vm.getApi()));
+    return this;
+  }
+
+  public ValueMappingModel setApiToImpl(List<ValueMapping> apiToImpl) {
+    apiToImpl.forEach(vm -> this.apiToImpl.put(vm.getApi(), vm.getImpl()));
+    return this;
+  }
+
+  public Optional<String> mapToImpl(String apiValue) {
+    return Optional.ofNullable(
+        apiToImpl.getOrDefault(
+            apiValue.toUpperCase(Locale.ROOT),
+            apiToImplBidirectional.getOrDefault(apiValue.toUpperCase(Locale.ROOT), null)));
+  }
+
+  public Optional<String> mapToApi(String implValue) {
+    return Optional.ofNullable(
+        implToApi.getOrDefault(implValue, implToApiBidirectional.getOrDefault(implValue, null)));
```

Comment:
The second statement could be simplified to "implToApiBidirectional.get(implValue)"

Reply:
ok